File: DrawIt/MyPen.vb
Imports System.ComponentModel
Imports System.Drawing.Drawing2D
Imports System.Runtime.CompilerServices

<Serializable>
Public Class MyPen
	Implements INotifyPropertyChanged, ICloneable

#Region "New"
	Sub New()
		PBrush.SolidColor = Color.Black
		PBrush.LInterpolate = True
		AddHandler PBrush.PropertyChanged, AddressOf NPC
	End Sub

	Private Sub NPC(sender As Object, e As PropertyChangedEventArgs)
		RaiseEvent PropertyChanged(Me, e)
	End Sub
#End Region

#Region "Event"
	<NonSerialized>
	Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

	Private Sub NotifyPropertyChanged(<CallerMemberName> Optional propertyName As String = "")
		RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
	End Sub

#End Region

#Region "Pen"
	Private _br As MyBrush = New MyBrush()
	Public Property PBrush() As MyBrush
		Get
			Return _br
		End Get
		Set(value As MyBrush)
			If Not value.Equals(_br) Then
				_br = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private _wid As Single = 1
	Public Property PWidth() As Single
		Get
			Return _wid
		End Get
		Set(value As Single)
			If Not value = _wid Then
				_wid = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private d_cap As DashCap = DashCap.Flat
	Public Property PDashCap() As DashCap
		Get
			Return d_cap
		End Get
		Set(value As DashCap)
			If Not value = d_cap Then
				d_cap = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private d_style As DashStyle = DashStyle.Solid
	Public Property PDashstyle() As DashStyle
		Get
			Return d_style
		End Get
		Set(value As DashStyle)
			If Not value = d_style Then
				d_style = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private st_cap As LineCap = LineCap.Flat
	Public Property PStartCap() As LineCap
		Get
			Return st_cap
		End Get
		Set(value As LineCap)
			If Not value = st_cap Then
				st_cap = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private end_cap As LineCap = LineCap.Flat
	Public Property PEndCap() As LineCap
		Get
			Return end_cap
		End Get
		Set(value As LineCap)
			If Not value = end_cap Then
				end_cap = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private l_join As LineJoin = LineJoin.Miter
	Public Property PLineJoin() As LineJoin
		Get
			Return l_join
		End Get
		Set(value As LineJoin)
			If Not value = l_join Then
				l_join = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private scale_x As Single = 1
	Public Property ScaleX() As Single
		Get
			Return scale_x
		End Get
		Set(value As Single)
			If Not value = scale_x Then
				scale_x = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property

	Private scale_y As Single = 1
	Public Property ScaleY() As Single
		Get
			Return scale_y
		End Get
		Set(value As Single)
			If Not value = scale_y Then
				scale_y = value
				NotifyPropertyChanged()
			End If
		End Set
	End Property
#End Region

#Region "Clone"
	''' <summary>
	''' Creates an exact copy of this <see cref="MyPen"/> object.
	''' </summary>
	Public Function Clone() As Object Implements ICloneable.Clone
		Dim _new As New MyPen
		For Each pd As PropertyDescriptor In TypeDescriptor.GetProperties(GetType(MyPen))
			pd.SetValue(_new, pd.GetValue(Me))
		Next
		_new.PBrush = PBrush.Clone
		Return _new
	End Function
#End Region

End Class

File: DrawIt.Models/Classes/MyPen.cs
using System.ComponentModel;
using System.Drawing;
using System.Runtime.CompilerServices;

namespace DrawIt.Models
{

	[Serializable]
	public class MyPen : INotifyPropertyChanged, ICloneable
	{
		private readonly PropertyChangedEventHandler _brushChangedHandler;

		public MyPen()
		{
			_brushChangedHandler = (sender, e) => PropertyChanged?.Invoke(this, e);
			var solid = (MySolidBrush)PBrush;
			solid.Color = Color.Black;
			PBrush.PropertyChanged += _brushChangedHandler;
		}

		#region INotifyPropertyChanged
		public event PropertyChangedEventHandler? PropertyChanged;

		private void NotifyPropertyChanged([CallerMemberName] string propertyName = "")
		{
			PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
		}
		#endregion

		#region ICloneable
		/// <summary>
		///		Creates an exact copy of this <see cref="MyPen"/> object.
		/// </summary>
		public object Clone()
		{
			MyPen _new = new();
			foreach (PropertyDescriptor pd in TypeDescriptor.GetProperties(typeof(MyPen)))
				pd.SetValue(_new, pd.GetValue(this));
			_new.PBrush = (MyBrush)PBrush.Clone();
			return _new;
		}
		#endregion

		
		#region Properties
		private MyBrush _br = new MySolidBrush();
		public MyBrush PBrush
		{
			get
			{
				return _br;
			}
			set
			{
				if (value is null || ReferenceEquals(value, _br)) return;
				_br.PropertyChanged -= _brushChangedHandler;
				_br = value;
				_br.PropertyChanged += _brushChangedHandler;
				NotifyPropertyChanged();
			}
		}

		private float _wid = 1;
		public float PWidth
		{
			get
			{
				return _wid;
			}
			set
			{
				if (!value.Equals(_wid))
				{
					_wid = value;
					NotifyPropertyChanged();
				}
			}
		}

		private string d_cap = "Flat";
		public string PDashCap
		{
			get
			{
				return d_cap;
			}
			set
			{
				if (!string.Equals(value, d_cap, StringComparison.Ordinal))
				{
					d_cap = value;
					NotifyPropertyChanged();
				}
			}
		}

		private string d_style = "Solid";
		public string PDashstyle
		{
			get
			{
				return d_style;
			}
			set
			{
				if (!string.Equals(value, d_style, StringComparison.Ordinal))
				{
					d_style = value;
					NotifyPropertyChanged();
				}
			}
		}

		private string st_cap = "Flat";
		public string PStartCap
		{
			get
			{
				return st_cap;
			}
			set
			{
				if (!string.Equals(value, st_cap, StringComparison.Ordinal))
				{
					st_cap = value;
					NotifyPropertyChanged();
				}
			}
		}

		private string end_cap = "Flat";
		public string PEndCap
		{
			get
			{
				return end_cap;
			}
			set
			{
				if (!string.Equals(value, end_cap, StringComparison.Ordinal))
				{
					end_cap = value;
					NotifyPropertyChanged();
				}
			}
		}

		private string l_join = "Miter";
		public string PLineJoin
		{
			get
			{
				return l_join;
			}
			set
			{
				if (!string.Equals(value, l_join, StringComparison.Ordinal))
				{
					l_join = value;
					NotifyPropertyChanged();
				}
			}
		}

		private float scale_x = 1;
		public float ScaleX
		{
			get
			{
				return scale_x;
			}
			set
			{
				if (!value.Equals(scale_x))
				{
					scale_x = value;
					NotifyPropertyChanged();
				}
			}
		}

		private float scale_y = 1;
		public float ScaleY
		{
			get
			{
				return scale_y;
			}
			set
			{
				if (!value.Equals(scale_y))
				{
					scale_y = value;
					NotifyPropertyChanged();
				}
			}
		}
		#endregion
	}

}
